src/video_core/renderer_software/renderer_software.cpp

```diff
@@ -2,18 +2,87 @@
 // Licensed under GPLv2 or any later version
 // Refer to the license.txt file included.
 
+#include "common/color.h"
+#include "core/core.h"
+#include "core/hw/gpu.h"
+#include "core/hw/hw.h"
+#include "core/hw/lcd.h"
 #include "video_core/renderer_software/renderer_software.h"
 
-namespace VideoCore {
+namespace Software {
 
 RendererSoftware::RendererSoftware(Core::System& system, Frontend::EmuWindow& window)
-    : VideoCore::RendererBase{system, window, nullptr},
-      rasterizer{std::make_unique<RasterizerSoftware>()} {}
+    : VideoCore::RendererBase{system, window, nullptr}, memory{system.Memory()},
+      rasterizer{std::make_unique<RasterizerSoftware>(system.Memory())} {}
 
 RendererSoftware::~RendererSoftware() = default;
 
 void RendererSoftware::SwapBuffers() {
+    PrepareRendertarget();
     EndFrame();
 }
 
-} // namespace VideoCore
+void RendererSoftware::PrepareRendertarget() {
+    for (int i : {0, 1, 2}) {
+        int fb_id = i == 2 ? 1 : 0;
```

Comment:
```suggestion
        const int fb_id = i == 2 ? 1 : 0;
```
also please check the other variables if they can be const, I won't comment on it again here